File: eu.openanalytics.phaedra.tutorials.scripting/3 RenameBarcode/renameBarcode.js
load("script://dc/common.js");

forEachReading(function(reading) {
	var barcode = reading.getBarcode();
	var parentFolderPath = API.get("FileUtils").getPath(reading.getSourcePath());
	var parentFolderName = API.get("FileUtils").getName(parentFolderPath);
	barcode = barcode + "_" + parentFolderName.substring(0, 4);
	reading.setBarcode(barcode);
});

File: eu.openanalytics.phaedra.tutorials.scripting/3 RenameBarcode/renameBarcode.py
import os

for reading in ctx.getReadings():
	barcode = reading.getBarcode()
	parentFolderPath = os.path.dirname(reading.getSourcePath())
	parentFolderName = os.path.basename(parentFolderPath)
	barcode = barcode + "_" + parentFolderName[0:4]
	reading.setBarcode(barcode)